### PR TITLE
extend Transportation

### PR DESCRIPTION
**Pull request by danielnaumau**

### CODE REVIEW

**Comment by mjanuszkiewicz-tt** [comment]
I'm confused about `RoundLimit` - it's an internal representation in the api, and not something that you can pass via JSON. There is no `roundLimit` field in the JSON message - it's `maxChanges`, which is already modeled.

File: src/main/java/com/traveltime/sdk/dto/common/transportation/DrivingTrain.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+@Value
+@Builder
+@Jacksonized
+@AllArgsConstructor
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class DrivingTrain implements Transportation {
+    @Positive(message = "ptChangeDelay must be greater than 0")
+    Integer ptChangeDelay;
+    @Positive(message = "drivingTimeLimit must be greater than 0")
+    Integer drivingTimeLimit;
```
Comment (bartosz-witkowski) [inline src/main/java/com/traveltime/sdk/dto/common/transportation/DrivingTrain.java:21]: In docs we have `driveTimeToStation`

File: src/main/java/com/traveltime/sdk/dto/requests/timefilterfast/transportation/DrivingAndPublicTransport.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.extern.jackson.Jacksonized;
+
+
+@Value
+@Builder
+@Jacksonized
+@AllArgsConstructor
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class DrivingAndPublicTransport implements Transportation {
+    @Positive(message = "walkingTime must be greater than 0")
```
Comment (bartosz-witkowski) [inline src/main/java/com/traveltime/sdk/dto/requests/timefilterfast/transportation/DrivingAndPublicTransport.java:18]: I don't think we expose those parameters in any api yet.

Comment (danielnaumau) [reply src/main/java/com/traveltime/sdk/dto/requests/timefilterfast/transportation/DrivingAndPublicTransport.java:18]: So I remove it everywhere, right?

Comment (bartosz-witkowski) [reply src/main/java/com/traveltime/sdk/dto/requests/timefilterfast/transportation/DrivingAndPublicTransport.java:18]: In `...timefilterfast.*`

File: src/main/java/com/traveltime/sdk/dto/requests/timefilterfast/Transportation.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.traveltime.sdk.dto.requests.timefilterfast.transportation.*;
+
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.NAME,
+    property = "type"
+)
+@JsonSubTypes({
```
Comment (mjanuszkiewicz-tt) [inline src/main/java/com/traveltime/sdk/dto/requests/timefilterfast/Transportation.java:11]: According to our API docs [1], time filter fast only supports `Possible values: public_transport, driving, driving+public_transport`.

I'm not sure which version is true though, @bartosz-witkowski ?

[1] https://docs.traveltime.com/api/reference/time-filter-fast#arrival_searches-many_to_one-transportation-type

Comment (bartosz-witkowski) [reply src/main/java/com/traveltime/sdk/dto/requests/timefilterfast/Transportation.java:11]: Right now we only have the four modes you mentioned but in the near future we'll be rolling out different modes.

Comment (mjanuszkiewicz-tt) [reply src/main/java/com/traveltime/sdk/dto/requests/timefilterfast/Transportation.java:11]: I mentioned three though :grimacing:

Comment (bartosz-witkowski) [reply src/main/java/com/traveltime/sdk/dto/requests/timefilterfast/Transportation.java:11]: Ermm derp. Also scratch what I said currently we support

-  `walking+ferry` (as a hack `walking` defaults to this)
- `cycling+ferry` (as a hack `cycling` defaults to this)
- `driving+ferry` (as a hack `driving` defaults to this)
- `driving+public_transport`
- `public_transport`

on the json api. For the proto API it should be the same.

Comment (mjanuszkiewicz-tt) [reply src/main/java/com/traveltime/sdk/dto/requests/timefilterfast/Transportation.java:11]: Ok, so this is fine, docs need to be updated.

**Review by mjanuszkiewicz-tt — APPROVED** [review]

**Review by ajakubauskas — APPROVED** [review]

File: src/main/java/com/traveltime/sdk/dto/common/transportation/DrivingTrain.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+@Jacksonized
+@AllArgsConstructor
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class DrivingTrain implements Transportation {
+    @Positive(message = "ptChangeDelay must be greater than 0")
+    Integer ptChangeDelay;
+    @Positive(message = "drivingTimeToStation must be greater than 0")
+    Integer drivingTimeToStation;
+    @Positive(message = "parkingTime must be greater than 0")
+    Integer parkingTime;
```
Comment (Donatas-L) [inline src/main/java/com/traveltime/sdk/dto/common/transportation/DrivingTrain.java:23]: Without looking at docs, it's hard to understand how to use these fields. I would suggest to comment these fields, preferably in javaDoc format.

Comment (danielnaumau) [reply src/main/java/com/traveltime/sdk/dto/common/transportation/DrivingTrain.java:23]: I'm not sure that javaDoc is well compatible with Lombok, so I'm planning to move all these params to readme.

**Review by Donatas-L — APPROVED** [review]

**Pull request merged**